msmart/device/AC/command.py

```diff
@@ -468,7 +473,11 @@ def _parse(self, payload: memoryview) -> None:
         # self.appliance_error = (payload[1] & 0x80) > 0
 
         # Unpack target temp and mode byte
-        self.target_temperature = (payload[2] & 0xF) + 16.0
+        self.target_temperature = payload[13] & 0x1F
+        if self.target_temperature != 0:

```

Comment:
Couple of things.
* I'd like to see the byte parsing occur "in order", so could you relocate the handling of payload[13] to around line 529
* I think it would be better to utilize a local variable when parsing the data, and then assign the final result to self.target_temperature
* The Lua code has an additional condition to handle wider ranges
```lua
local temp = bit.band(messageBytes[13], 0x1F)
if (temp > 0 and temp <= 25)  then 
  keyP["temperature"] = temp + 12 
elseif(temp == 0)then
else
  keyP["temperature"] = temp - 19
end
```